### PR TITLE
Fixes #493: compatibility with plugin "Enable Media Replace" broken with WP 5.3

### LINKED ISSUE #493: Compatibility with plugin "Enable Media Replace" broken with WP 5.3
In its version 3.3.8, the plugin Enable Media Replace changed some things to adapt to WordPress 5.3’s new upload process.
Since this version, our compatibility is broken. Currently, Imagify is hooking `'emr_unfiltered_get_attached_file'` to initiate everything:
```php
add_filter( 'emr_unfiltered_get_attached_file', [ \Imagify\ThirdParty\EnableMediaReplace\Main::get_instance(), 'init' ] );
```
But now in `\EnableMediaReplace\Replacer::__construct()`:
```php
      if (function_exists('wp_get_original_image_path')) // WP 5.3+
      {
          $source_file = wp_get_original_image_path($post_id);
          if ($source_file === false) // if it's not an image, returns false, use the old way.
            $source_file = trim(get_attached_file($post_id, apply_filters( 'emr_unfiltered_get_attached_file', true )));
      }
      else
        $source_file = trim(get_attached_file($post_id, apply_filters( 'emr_unfiltered_get_attached_file', true )));
```
If `wp_get_original_image_path()` doesn’t return `false`, the hook `'emr_unfiltered_get_attached_file'` is not triggered anymore.

This results in the images being improperly optimized, and the old backup file + old webp images not being replaced.
[Plugin](https://wordpress.org/plugins/enable-media-replace/).

### PR DESCRIPTION
**Pull request by Screenfeed**
This fixes our compatibility with the plugin "Enable Media Replace". It was broken since EMR updated its code to adapt to WordPress 5.3’s new upload process.

Sum up:
- Use the hook `'wp_handle_replace'` to initiate the class, instead of `'emr_unfiltered_get_attached_file'`.
- Removed some `if` statements and tests in `init()` that are not needed anymore.
- The old webp file paths are stored in `init()` now: like explained in the DocBlock, this prevents some orphan files from being kept indefinitely when the user chooses not to rename the files.
- Deprecate the method `store_old_backup_path()`, as it is not needed anymore: the old backup path is already correct and the old webp file paths are stored in `init()`.

Fixes #493.

Tests done locally. I will explain the procedure to the QA team.

### CODE REVIEW

File: inc/3rd-party/enable-media-replace/classes/Main.php [hunk 1]
```diff
@@ -1,174 +1,119 @@
...
+		 * - Do not rename the files: the thumbnails may still get new names because of the suffix containing the image dimensions, which may differ (for example when thumbnails are scaled, not cropped).
+		 * In this last case, the thumbnails with the old dimensions are removed from the drive and from the WP’s post meta, so there is no need of keeping orphan webp files that would stay on the drive for ever, even after the attachment is deleted from WP.
+		 */
+		$media_files = $this->process->get_media()->get_media_files();
+
+		if ( $media_files ) {
+			foreach ( $media_files as $media_file ) {
+				$this->old_webp_paths[] = imagify_path_to_webp( $media_file['path'] );
+			}
+		}
```
[the remark this answers is not in the document]
Comment (Screenfeed) [reply inc/3rd-party/enable-media-replace/classes/Main.php:100]: Actually, typecasting is not even required, the method [always returns an array](https://github.com/wp-media/imagify-plugin/blob/d81c3c2078eceb5c139966e0402f286944d98912/classes/Media/MediaInterface.php#L305-L314). I will simply remove the `if` statement.

**Comment by Screenfeed** [comment]
@wp-media/dev-qa Ready for QA.
[imagify-1.9.10-alpha1-emrfix.zip](https://github.com/wp-media/imagify-plugin/files/4622312/imagify-1.9.10-alpha1-emrfix.zip)

Set up for the test:
- Go to Imagify’s settings:
  - Make sure "Backup original images" is enabled, so you will be able to restore later.
  - Also, enable "Create webp versions of images".
- You will need 2 images, let's call them "Control A" and "Control B". Those are the images we will use as references.
  - Upload "Control A" and "Control B" to your site and optimize them with Imagify.
  - Keep a copy of them not optimized, you will have to upload them later again.
  - Both of them need to be successfully optimized, not "Already optimized" (pick another image if it’s the case).
  - You need to be able to see the uploaded files directly and their size (ftp).
- Upload the image "Control A" again and optimize it with Imagify. Let’s call it "Test image".
- Install and activate [Enable Media Replace](https://wordpress.org/plugins/enable-media-replace/).
- Go to your media library and enable the list display.

The test:
- "Test image": click the "Replace media" link from the media list (it's available for each media) or "Upload a new file" in the media edition page (in the column on the right).
- Choose the same image you used for "Control B" as replacement. Select "Replace the file, use new file name and update all links". Click "Upload".
- Go to your file browser and verify:
  - "Test image" and its thumbnails should be replaced by "Control B".
  - Their filename should have changed.
  - When Imagify completes its work:
    - The webp files should also be replaced by "Control B versions".
    - Compare the file size of these "Test image" files with the size of "Control B" files: you should have the same thumbnails with the same sizes.
    - Look at the backup file in `/wp-content/uploads/backup/2020/05`, it should also contain a "Test image" identical to the "Control B" in the same folder. The backup file should also be renamed.
- Back to the media library, same procedure, still with "Test image": click the "Replace media" link.
- Choose the same image you used for "Control A" as replacement this time. Keep "Just replace the file" selected. Click "Upload".
- Go to your file browser and verify:
  - "Test image" and its thumbnails should be replaced by "Control A".
  - Their filename hasn't changed (except for the dynamic part with the dimensions, it may change).
  - When Imagify completes its work:
    - The webp files should also be replaced by "Control A versions".
    - Compare the file size of these "Test image" files with the size of "Control A" files: you should have the same thumbnails with the same sizes.
    - Look at the backup file in `/wp-content/uploads/backup/2020/05`, it should also contain a "Test image" identical to the "Control A" in the same folder. The backup file should not be renamed.
- Back to the media library, restore "Test image" by clicking "Restore Original" in the Imagify’s column.
- Click the "Replace media" link.
- Choose the same image you used for "Control B" as replacement this time. We don't care about the option for changing the file name. Click "Upload".
- Imagify should not automatically optimize "Test image".

**Comment by markonikolic985** [comment]
Tested this on WP v5.4.1 and PHP 7.3 installation:

✅"Test image" and its thumbnails should be replaced by "Control B".
✅Their filename should have changed.

When Imagify completes its work:

**❌The webp files should also be replaced by "Control B versions".**

- ControlB-1 does not have a backup and WebP (although it says WebP is generated) - [screenshot1](https://www.dropbox.com/s/q1otb0ha71vzcss/screen%20shot%202020-05-14%20at%2010.52.21%20am.png?dl=0)

- WebP is missing (refreshed FileZilla few times to be 100% sure) - [screenshot2](https://www.dropbox.com/s/n9qde3y553rtxy6/screen%20shot%202020-05-14%20at%2010.54.00%20am.png?dl=0)


**❌Compare the file size of these "Test image" files with the size of "Control B" files: you should have the same thumbnails with the same sizes.**

- Not the same size - [screenshot3](https://www.dropbox.com/s/0vv7aqh7wbetpnz/screen%20shot%202020-05-14%20at%2011.02.09%20am.png?dl=0)

**❌Look at the backup file in /wp-content/uploads/backup/2020/05, it should also contain a "Test image" identical to the "Control B" in the same folder. The backup file should also be renamed.**

- backup is missing - [screenshot4](https://www.dropbox.com/s/tqfjoh96luwry2x/screen%20shot%202020-05-14%20at%2011.04.35%20am.png?dl=0)


Tested again with another set of images but it's still the same. Stopped at this point, let's see what could go wrong here.

**Comment by Screenfeed** [comment]
@markonikolic985 Here is a new version that should work this time:
[imagify-1.9.10-alpha3-emrfix.zip](https://github.com/wp-media/imagify-plugin/files/4629448/imagify-1.9.10-alpha3-emrfix.zip)

**Comment by markonikolic985** [comment]
@Tabrisrp 

Tested this. 

Everything works as expected. 

Failed points from my previous message are fixed so I was able to proceed with further steps.

All these worked as expected.  ✅

**Pull request merged**